salt/config/__init__.py

```diff
@@ -3270,6 +3274,9 @@ def get_id(opts, cache_minion_id=False):
                   .format(os.path.join(salt.syspaths.CONFIG_DIR, 'minion')))
 
     newid = salt.utils.network.generate_minion_id()
+
+    if opts.get('minion_id_lowercase', True):
```

Comment:
I'm not sure we should be defaulting this to `True`. Even though it's in the config, I would feel more comfortable here if we either raised a `KeyError` if it wasn't set or just let it default back to `None`.